python/idsse_common/idsse/common/grid_proj.py

```diff
@@ -0,0 +1,113 @@
+"""Module that wraps pyproj (cartographic/geodetic) library and transforms objects into other data forms"""
+# -------------------------------------------------------------------------------
+# Created on Mon Jul 31 2023
+#
+# Copyright (c) 2023 Colorado State University. All rights reserved. (1)
+#
+# Contributors:
+#     Mackenzie Grimes (1)
+#
+# -------------------------------------------------------------------------------
+# pylint: disable=invalid-name
+
+from typing import Self, Tuple, Union, Any, Optional
+from enum import Enum, auto
+from math import floor
+
+from pyproj import CRS, Transformer
+from pyproj.enums import TransformDirection
+
+
+class PixelRounding(Enum):
+    """Transformations to apply to calculated pixels when casting to ints"""
+    ROUND = auto()
+    FLOOR = auto()
+
+
+class GridProj:
+    """Wrapper for pyproj instance with methods to convert to and from geographic coordinates, pixels, etc."""
+    def __init__(self,
+                 crs: CRS,
+                 latitude: float,
+                 longitude: float,
+                 width: float,
+                 height: float,
+                 dx: float,
+                 dy: Optional[float] = None):
+        # pylint: disable=too-many-arguments,unpacking-non-sequence
+        self._trans = Transformer.from_crs(crs.geodetic_crs, crs)
+        self._x_offset, self._y_offset = self._trans.transform(longitude, latitude)
+        self._w = width
+        self._h = height
+        self._dx = dx
+        self._dy = dy if dy else dx
+
+    @classmethod
+    def from_proj_grid_spec(cls, proj_string: str, grid_string: str) -> Self:
+        """ Create GridProj instance from projection grid specs
+
+        Args:
+            proj_string (str): pyproj projection string
+            grid_string (str): Grid string
+
+        Returns:
+            Self: New instance of GridProj which can then be converted to any format
+        """
+        crs = CRS.from_proj4(proj_string)
+        grid_args = {
+            key[1:]: float(value)
+            for key, value in (
+                pair.split('=') for pair in grid_string.split(' ')
+            )
+        }
+        return GridProj(crs,
+                        grid_args['lat_ll'], grid_args['lon_ll'],
+                        int(grid_args['w']), int(grid_args['h']),
+                        grid_args['dx'], grid_args['dy'])
+
+    def map_proj_to_pixel(self, x, y) -> Tuple[float, float]:
```

Comment:
It would be best of all _to_pixel functions supported rounding, so add the same logic to map_proj_to_pixel() as map_geo_to_pixel() has. It might be good to create a private function to round so that both _to_pixel functions can call the same code.